dynamic/dynamic_message_test.go

```diff
@@ -1861,6 +1861,48 @@ func loadExtension(t *testing.T, ed *proto.ExtensionDesc) *desc.FieldDescriptor
 	return fd
 }
 
+func TestSetGetOneOfDefaultValue(t *testing.T) {
+	fd, err := desc.LoadFileDescriptor("desc_test_oneof.proto")
+	testutil.Ok(t, err)
+	md := fd.FindSymbol("testprotos.Value").(*desc.MessageDescriptor)
+
+	var testCases = []struct {
+		fieldName  string
+		fieldValue interface{}
+	}{
+		{"binary_value", []byte{}},
+		{"string_value", ""},
+		{"boolean_value", false},
+		{"int_value", int32(0)},
+		{"int64_value", int64(0)},
+		{"double_value", float64(0)},
+		{"float_value", float32(0)},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.fieldName, func(t *testing.T) {
+			dm := NewMessage(md)
```

Comment:
Let's construct the message outside the loop. That way, we are also testing that a call to `SetField` is properly clear the others in the one-of. (As is, it is always a new message with no other fields set.)

To that end, you can also move the section below, that clears the field, outside of the loop. (It's fine to just remember the last field set by moving the definition of `field` to outside the loop, too.